simulator.py

```diff
@@ -25,11 +26,22 @@ def expand_single_gate(gate, i, num_qbits):
 
 
 def expand_double_gate(gate, i, j, num_qbits):
-    assert j == i + 1, "Operation only implemented for j == i + 1"
-    left_bits = i
-    right_bits = num_qbits - 2 - i
+    assert i < j, "Operation only implemented for i < j"
 
-    return (Identity ** left_bits) * gate * (Identity ** right_bits)
+    k = j - 1
+    left_bits = k
+    right_bits = num_qbits - 2 - k
+
+    operator = (Identity ** left_bits) * gate * (Identity ** right_bits)
+
+    # Swap left bit until it is left of right bit before real gate operator
+    # and swap it left again after real gate operator
+    while k > i:
+        swap_operator = expand_double_gate(SWAP, k-1, k, num_qbits)
+        operator = swap_operator | operator | swap_operator
+        k -= 1
```

Comment:
The overall idea looks correct but the risk of getting a one-off error is big. We should add a unit test of some kind.

Reply:
added a unittest :)